animation_nodes/nodes/gpencil/gp_frame_info.py

```diff
@@ -0,0 +1,68 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import GPStroke
+from ... base_types import AnimationNode, VectorizedSocket
+
+strokeTypeItems = [
+    ("ALL", "All Strokes", "Get all strokes", "NONE", 0),
+    ("INDEX", "Index Stroke ", "Get a specific stroke", "NONE", 1)
+]
+
+class GPFrameInfoNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPFrameInfoNode"
+    bl_label = "GP Frame Info"
+    errorHandlingType = "EXCEPTION"
+
+    strokeType: EnumProperty(name = "Stroke Type", default = "ALL",
+        items = strokeTypeItems, update = AnimationNode.refresh)
+
+    useIntegerList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("GPFrame", "Frame", "frame", dataIsModified = True)
+
+        if self.strokeType == "ALL":
+            self.newOutput("GPStroke List", "Strokes", "strokes")
+        elif self.strokeType == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useIntegerList",
+            ("Stroke Index", "strokeIndex"), ("Stroke Indices", "strokeIndices")))
+
+            self.newOutput(VectorizedSocket("GPStroke", "useIntegerList",
+            ("Stroke", "stroke"), ("Strokes", "strokes")))
+        self.newOutput("Float", "Frame Number", "frameNumber")
+
+    def draw(self, layout):
+        layout.prop(self, "strokeType", text = "")
+
+    def getExecutionFunctionName(self):
+        if self.strokeType == "ALL":
+            return "execute_AllStrokes"
+        elif self.strokeType == "INDEX":
+            if self.useIntegerList:
+                return "execute_StrokeIndices"
+            else:
+                return "execute_StrokeIndex"
+
+    def execute_AllStrokes(self, frame):
+        return frame.strokes, frame.frameNumber
+
+    def execute_StrokeIndex(self, frame, strokeIndex):
+        strokes = frame.strokes
+        if len(frame.strokes) == 0:
+            return GPStroke(), frame.frameNumber
+        return self.getStroke(strokes, strokeIndex), frame.frameNumber
+
+    def execute_StrokeIndices(self, frame, strokeIndices):
+        strokes = frame.strokes
+        outStrokes = []
+        if len(frame.strokes) == 0:
+            return outStrokes, frame.frameNumber
+
+        for index in strokeIndices:
+            outStrokes.append(self.getStroke(strokes, index))
+        return outStrokes, frame.frameNumber
+
+    def getStroke(self, strokes, strokeIndex):
+        try: return strokes[strokeIndex]
```

Comment:
Don't use exceptions if you can write defensive code. In this case, you can easily check if the index indeed exist.

Reply:
Okay.